ios/utils/UIView+RNSUtility.h

```diff
@@ -0,0 +1,23 @@
+#import <Foundation/Foundation.h>
+
+#ifdef RCT_NEW_ARCH_ENABLED
+#import <React/RCTSurfaceTouchHandler.h>
+#else
+#import <React/RCTTouchHandler.h>
+#endif // RCT_NEW_ARCH_ENABLED
+
+#ifdef RCT_NEW_ARCH_ENABLED
+#define RNS_TOUCH_HANDLER_ARCH_TYPE RCTSurfaceTouchHandler
+#else
+#define RNS_TOUCH_HANDLER_ARCH_TYPE RCTTouchHandler
```

Comment:
I tried using `typedef`s instead of macros here, but stumbled upon [issues with nullability](https://developer.apple.com/swift/blog/?id=25) (section about "audited regions") and had unexpected issues doing any casting for typedef'd types -> went for macros.